internal/output/reporter.go

```diff
@@ -9,24 +9,24 @@ import (
 )
 
 type Reporter struct {
-	stdout       io.Writer
-	stderr       io.Writer
-	outputAsJSON bool
+	stdout io.Writer
+	stderr io.Writer
+	format string
 }
 
-func NewReporter(stdout io.Writer, stderr io.Writer, outputAsJSON bool) *Reporter {
+func NewReporter(stdout io.Writer, stderr io.Writer, format string) *Reporter {
```

Comment:
I generally prefer enums for these arguments, but I'm not familiar enough with Go to know if that's practical or not here, Go's enum seems very clunky.

Reply:
Same - the main practical reason I didn't go with one here is that it'd require a cast since there's no way of having the cli library return an enum instead of a string; that's not the end of the world (and really what Go wants you to do in the first place), but combined with the fact that there are currently only a couple of supported formats I felt it would be best to keep it simple for now.